src/utils/PublicPathUtils.ts

```diff
@@ -9,16 +9,18 @@ export class PublicPathUtils {
   // Fallback on last parsed script if document.currentScript is not available.
   public static detectPublicPath() {
     if (!this.pathHasBeenConfigured) {
-      let path;
-      let currentScript = DomUtils.getCurrentScript();
-      if (Utils.isNullOrUndefined(currentScript)) {
-        let scripts = document.getElementsByTagName('script');
-        path = this.parseScriptDirectoryPath(scripts[scripts.length - 1]);
-      } else {
-        let script = currentScript;
-        path = this.parseScriptDirectoryPath(script);
-      }
-      __webpack_public_path__ = path;
+      __webpack_public_path__ = PublicPathUtils.getDynamicPublicPath();
+    }
+  }
+
+  public static getDynamicPublicPath() {
```

Comment:
Since this would now be needed to stay available under the same name and function signature going forward (since you guys will always need it), the best way to ensure that is to add public documentation in the class and the `getDynamicPublicPath`.

What gets documented publicly is really a big "warning" flag for us that we must keep it for backward compatibility and take great care if we need to change it.

Otherwise someone might assume it's only used internally.